python/cudf/utils/cudautils.py

```diff
@@ -610,6 +610,33 @@ def gpu_diff(in_col, out_col, N):
             out_col[i] = -1
 
 
+@cuda.jit
+def gpu_round(in_col, out_col, decimal):
+    i = cuda.grid(1)
+    round_val = 10 ** (-1.0 * decimal)
+
+    if i < in_col.size:
+        newval = in_col[i] // round_val * round_val
+        remainder = fmod(in_col[i], round_val)
+
+        if remainder > (.5 * round_val) and in_col[i] > 0:
+            newval = newval + round_val
+            out_col[i] = newval
+
+        elif abs(remainder) < (.5 * round_val) and in_col[i] < 0:
+            newval = newval + round_val
+            out_col[i] = newval
+
+        else:
+            out_col[i] = newval
+
+
+def apply_round(data, decimal):
+    output_dary = rmm.device_array_like(data)
+    gpu_round.forall(output_dary.size)(data, output_dary, decimal)
```

Comment:
Need to check that `output_dary.size > 0` before triggering this kernel call, otherwise you can have a 0 sized kernel launch and explode.

Reply:
Added a check to avoid an angry CUDA error.